kura/org.eclipse.kura.nm/src/main/java/org/eclipse/kura/nm/NMConfigurationEnforcementHandler.java

```diff
@@ -0,0 +1,54 @@
+/*******************************************************************************
+ * Copyright (c) 2023 Eurotech and/or its affiliates and others
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *  Eurotech
+ *******************************************************************************/
+package org.eclipse.kura.nm;
+
+import java.util.Objects;
+
+import org.freedesktop.dbus.exceptions.DBusException;
+import org.freedesktop.dbus.interfaces.DBusSigHandler;
+import org.freedesktop.networkmanager.Device;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class NMConfigurationEnforcementHandler implements DBusSigHandler<Device.StateChanged> {
+
+    private static final Logger logger = LoggerFactory.getLogger(NMConfigurationEnforcementHandler.class);
+    private final NMDbusConnector nm;
+
+    public NMConfigurationEnforcementHandler(NMDbusConnector nmDbusConnector) {
+        this.nm = Objects.requireNonNull(nmDbusConnector);
```

Comment:
Didn't you suggest me to use the NMDbusConnector.getInstance in place of passing the NMDbusConnector reference ?

Reply:
Yes ... and I was absolutely ✨**wrong**✨ !

I started my implementation like that and then realised how wrong I was when implementing tests.

Sorry 🙏